AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/managespace/ManageSpaceFragment.kt

```diff
@@ -0,0 +1,386 @@
+                setTitle("Delete unused media files")
+                setMultiChoiceItems(unusedFileNames, CheckedItems.All)
+                setPositiveButton(R.string.dialog_positive_delete_selected)
+                setNegativeButton(R.string.dialog_cancel)
+            }
+
+            if (deleteFilesPromptResult is DialogResult.Ok.MultipleChoice) {
+                val checkedItems = deleteFilesPromptResult.checkedItems
+                val filesNamesToDelete = unusedFileNames.filterIndexed { index, _ -> checkedItems[index] }
+
+                withProgress(R.string.delete_media_message) {
+                    viewModel.deleteMedia(filesNamesToDelete)
+                }
+            }
+        } else {
+            showSnackbarIfCalculatingOrError(size)
+        }
+    }
+
+    /*************************************** Delete backups ***************************************/
+
+    context(CoroutineScope) private suspend fun onDeleteBackupsClick() {
+        val size = viewModel.flowOfDeleteBackupsSize.value
+        if (size is Size.FilesAndBytes) {
+            val formatter = LocalizedUnambiguousBackupTimeFormatter()
+            val backupFiles = size.files
+            val backupNames = backupFiles.map { formatter.getTimeOfBackupAsText(it) }
+
+            val chooseBackupsPromptResult = requireContext().awaitDialog {
+                setTitle("Delete backups")
+                setMultiChoiceItems(backupNames, CheckedItems.None)
+                setPositiveButton(R.string.dialog_positive_delete_selected)
+                setNegativeButton(R.string.dialog_cancel)
+            }
+
+            if (chooseBackupsPromptResult is DialogResult.Ok.MultipleChoice) {
+                val checkedItems = chooseBackupsPromptResult.checkedItems
+                val backupsToDelete = backupFiles.filterIndexed { index, _ -> checkedItems[index] }
+
+                withProgress("Deleting backups…") {
+                    viewModel.deleteBackups(backupsToDelete)
+                }
+            }
+        } else {
+            showSnackbarIfCalculatingOrError(size)
+        }
+    }
+
+    /************************************* Delete collection **************************************/
+
+    // TODO: Finish other AnkiDroid activities when the collection is deleted.
+    //   Note that this might be not quite trivial, as the activities might be visible to user.
+    //   One way would be to have the activities register broadcast receivers that perform finish;
+    //   Another would be maintaining weak references to them. Would be nice to find a better way.
+    context(CoroutineScope) private suspend fun onDeleteCollectionClick() {
+        val size = viewModel.flowOfDeleteCollectionSize.value
+        if (size is Size.Bytes) {
+            val deleteCollectionPromptResult = requireContext().awaitDialog {
+                setTitle("Delete collection 😱")
+                setMessage("Are you sure you want to delete your collection, media, and backups?")
```

Comment:
This one confuses me a little - message is deleting everything (col, media, backups) but title is just delete collection

if this truly deletes everything shouldn't it be more like "delete all app data" ? :thinking: 

```suggestion
                setTitle("Delete collection")
                setMessage("Are you sure you want to delete your collection, media, and backups?")
```

Reply:
Settings, app cache, etc is not deleted.

Reply:
Hmm is there some word that captures "more than collection" but "less than everything". "App Data" maybe? I don't like that. Delete collection seems fine fine though, go with it - resolving except extraction needed of course

Reply:
Regarding the “Clear storage” action on Android, which appears to be deleting “User data” and “cache”. Depending on where the collection is located, we call that either:

* Delete everything (Delete collection, media, backups, and settings)
* Delete app data (Delete settings and other app data), or

The former minus the latter is

* Delete collection (Delete collection, media, and backups. App settings are not deleted)

Media files are collection, I guess? Maybe backups aren't, but I can't imagine a scenario where I'd want to delete the collection but not its backups.